riscv/src/asm.rs

```diff
@@ -7,58 +7,87 @@ macro_rules! instruction {
         pub unsafe fn $fnname() {
             #[cfg(any(target_arch = "riscv32", target_arch = "riscv64"))]
             core::arch::asm!($asm, $($options)*);
-            #[cfg(not(any(target_arch = "riscv32", target_arch = "riscv64")))]
-            unimplemented!();
         }
     );
     ($(#[$attr:meta])*, $fnname:ident, $asm:expr, $($options:tt)*) => (
         $(#[$attr])*
         #[inline(always)]
         pub fn $fnname() {
             #[cfg(any(target_arch = "riscv32", target_arch = "riscv64"))]
-            unsafe { core::arch::asm!($asm, $($options)*) };
-            #[cfg(not(any(target_arch = "riscv32", target_arch = "riscv64")))]
-            unimplemented!();
+            unsafe { core::arch::asm!($asm, $($options)*)};
         }
-    );
+    )
 }
 
 instruction!(
-    /// `nop` instruction wrapper
+    /// `NOP` instruction wrapper
     ///
     /// The `NOP` instruction does not change any architecturally visible state, except for
-    /// advancing the pc and incrementing any applicable performance counters.
+    /// advancing the PC and incrementing any applicable performance counters.
     ///
     /// This function generates a no-operation; it's useful to prevent delay loops from being
     /// optimized away.
-    , nop, "nop", options(nomem, nostack));
+    ///
+    /// # Safety
+    ///
+    /// This function is always safe to call as it performs no operation.
```

Comment:
Not sure about the point of adding a safety section in completely safe functions